### PR TITLE
Move Validation Logic into PropertyValidator Class #149

### PR DESCRIPTION
**Pull request by alexsuperdev**
### This PR addresses:

#149

### I have verified that:

- [x] All related unit tests have been updated/created
- [x] Integration tests are passing

### Additional Notes

I created RequestValidator class that contains the logic to test all kind of requests.
The methods throw the ValidationException so that can be catched by called class like UserResource.

### CODE REVIEW

File: application/src/main/java/com/sanction/thunder/validation/RequestValidator.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+/**
+ * Provides validation logic for request validation for
+ * {@link com.sanction.thunder.models.User User} properties.
+ */
+public class RequestValidator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RequestValidator.class);
+
+  private PropertyValidator propertyValidator;
```
[the remark this answers is not in the document]
Comment (alexsuperdev) [reply application/src/main/java/com/sanction/thunder/validation/RequestValidator.java]: I don't think that final is needed here

File: application/src/main/java/com/sanction/thunder/validation/RequestValidator.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  }
+
+  /**
+   * Determines if user password and email are valid by given strings.
+   * The difference to validate(String password, String email) 
+   * is that log messages are different here
+   * @param password The password to test for validity
+   * @param email The email to test for validity
+   */
+  public void validateDelete(String password, String email) {
```
[the remark this answers is not in the document]
Comment (alexsuperdev) [reply application/src/main/java/com/sanction/thunder/validation/RequestValidator.java]: Good idea:
 private void validate(String password, String email, boolean isDelete) {
    if (email == null || email.isEmpty()) {
      if (isDelete) {
        LOG.warn("Cannot operate on a null user..");
      }
      else {
        LOG.warn("Cannot operate on user {} without a password.", email);
      }
      throw new ValidationException("Incorrect or missing email query parameter.");
    }

    if (password == null || password.isEmpty()) {
      LOG.warn("Attempted to get user {} without a password", email);
      throw new ValidationException("Incorrect or missing header credentials.");
    }
  }
so?

File: .settings/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/org.eclipse.core.resources.prefs
```
Comment (RohanNagar) [inline .settings/.gitignore]: Can you please remove all of these `.gitignore` files? If you need some additional files ignored, please put them in the top level `.gitignore` file.

**Comment by RohanNagar** [comment]
@alexsuperdev please completely delete these 4 files:

```
.settings/.gitignore
api/.settings/.gitignore
application/.settings/.gitignore
client/.settings/.gitignore
```

I still see all 4 files there. Once those are gone this will be good to go

**Comment by RohanNagar** [comment]
Thank you so much for the contribution @alexsuperdev! Really great work.

**Comment by alexsuperdev** [comment]
No problems, finally its done :)

**Pull request merged**